lib/Frontend/Frontend.cpp

```diff
@@ -420,11 +413,13 @@ shouldImplicityImportSwiftOnoneSupportModule(CompilerInvocation &Invocation) {
   if (Invocation.getSILOptions().shouldOptimize())
     return false;
 
-  if (shouldImportSwiftOnoneModuleIfNoneOrImplicitOptimization(
-          Invocation.getFrontendOptions().RequestedAction)) {
-    return true;
-  }
-  return Invocation.getFrontendOptions().isCreatingSIL();
+  // If we are not going through the SIL Optimizer with the
+  // given frontend action, do not load SwiftOnoneSupport.
+  //
+  // This optimization is disabled by -track-system-dependencies to preserve
+  // the explicit dependency.
+  return Invocation.getFrontendOptions().TrackSystemDeps
+      || Invocation.getFrontendOptions().isCreatingSIL();
```

Comment:
Strictly speaking you've turned on imports for SILGen too. Then again, the old logic would have missed EmitSIB and all the various IR-based options, so this is probably more future-proof. Still, it might slow down the tests; it may be worth special-casing SILGen to not do the import.

Reply:
Hm, this is the only use of this accessor.  I'll make it stricter.